src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -656,7 +667,9 @@ CHIP_ERROR OTARequestor::ExtractUpdateDescription(const QueryImageResponseDecoda
 
     VerifyOrReturnError(response.imageURI.HasValue(), CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorOnFailure(bdx::ParseURI(response.imageURI.Value(), nodeId, fileDesignator));
-    update.imageURI = response.imageURI.Value();
+    VerifyOrReturnError(IsSpanUsable(fileDesignator), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
It can't happen if `ParseURI` return `CHIP_NO_ERROR`.

Reply:
But I don't see any kind of checks when getting the SubSpan for file designator: https://github.com/project-chip/connectedhomeip/blob/master/src/protocols/bdx/BdxUri.cpp#L55

Reply:
I suppose I can move this check into ParseURI at the end before we return?

Reply:
There's a check for the length of the URI which implies that the span must be non-empty. IIRC we have a unit test for that.